packages/gatsby/src/commands/build.ts

```diff
@@ -76,16 +80,15 @@ module.exports = async function build(program: IBuildArgs): Promise<void> {
     graphqlTracing: program.graphqlTracing,
   })
 
-  const {
-    processPageQueries,
-    processStaticQueries,
-  } = queryUtil.getInitialQueryProcessors({
+  const { queryIds } = await calculateDirtyQueries({ store })
+
+  await runStaticQueries({
+    queryIds,
     parentSpan: buildSpan,
+    store,
     graphqlRunner,
   })
```

Comment:
There is no need to leak pageQueryIds into staticQueries
```suggestion
  await runStaticQueries({
    queryIds: queryIds.staticQueryIds,
    parentSpan: buildSpan,
    store,
    graphqlRunner,
  })
```